gtfs/spec/en/reference.md

```diff
@@ -373,14 +387,20 @@ To process the cost of a leg:
 1. The file `fare_leg_rules.txt` must be filtered by the fields that define the characteristics of travel, these fields are:
     - `fare_leg_rules.network_id`
     - `fare_leg_rules.from_area_id`
-    - `fare_leg_rules.to_area_id`<br/>
+    - `fare_leg_rules.to_area_id`
+    - `fare_leg_rules.from_timeframe_id`
+    - `fare_leg_rules.to_timeframe_id`
+    - `fare_leg_rules.service_id`<br/>    
 <br/>
 
 2. If the leg exactly matches a record in `fare_leg_rules.txt` based on the characteristics of travel, that record must be processed to determine the cost of the leg.
-3. If no exact matches are found, then empty entries in `fare_leg_rules.network_id`, `fare_leg_rules.from_area_id`, and `fare_leg_rules.to_area_id` must be checked to process the cost of the leg:
+3. If no exact matches are found, then empty entries in `fare_leg_rules.network_id`, `fare_leg_rules.from_area_id`, `fare_leg_rules.to_area_id`, `fare_leg_rules.from_timeframe_id`, `fare_leg_rules.to_timeframe_id`, and `fare_leg_rules.service_id` must be checked to process the cost of the leg:
     - An empty entry in `fare_leg_rules.network_id` corresponds to all networks defined in `routes.txt` excluding the ones listed under `fare_leg_rules.network_id`
     - An empty entry in `fare_leg_rules.from_area_id` corresponds to all areas defined in `areas.area_id` excluding the ones listed under `fare_leg_rules.from_area_id`
-    - An empty entry in `fare_leg_rules.to_area_id` corresponds to all areas defined in `areas.area_id` excluding the ones listed under `fare_leg_rules.to_area_id`<br/>
+    - An empty entry in `fare_leg_rules.to_area_id` corresponds to all areas defined in `areas.area_id` excluding the ones listed under `fare_leg_rules.to_area_id`
+    - An empty entry in `fare_leg_rules.from_timeframe_id` corresponds to all timeframes defined in `timeframes.timeframe_id` excluding the ones listed under `fare_leg_rules.from_timeframe_id`
```

Comment:
Can someone elaborate on the meaning "excluding the ones listed under"?

Reply:
@skinkie "excluding the ones listed under" means "excluding the unique list of values defined in all `fare_leg_rules.from_timeframe_id` records.
This design looks like this causes problems, as flagged by @bdferris-v2 in this doc: https://bit.ly/gtfs-fares-v2-fare-leg-rules-scope-proposal.

Reply:
I'll echo my concern from my [previous comment](https://github.com/google/transit/pull/357#issuecomment-1325447530) - I think this style of empty-value semantics (aka empty value => all values excluding explicitly mentioned values) is problematic and is just going to lead to trouble later on down the road.

I think for the timeframe proposal, I'd stick to "empty value matches everything" default semantics and rely on the fact that the ITO World timeframe proposal allows more flexibility in specifying complex timeframes when an explicit match is needed.

Reply:
We've discussed changing this behavior to: an empty value in `from_timeframe_id` or `to_timeframe_id` means that the start or end time doesn't affect the price. 
I think it would make more sense, but this goes against the empty semantics of the `from_area_id` and `to_area_id` fields.

Reply:
@bdferris-v2 "empty value matches everything" or "empty value means start or end time doesn't affect the price?"

Reply:
Assuming `from_timeframe_id` and `to_timeframe_id` (whatever the end up being called) stay in `fare_leg_rules.txt`, to me it's about how they affect whether a particular row in `fare_leg_rules.txt` matches a particular leg.  In that world, I'd prefer that an empty value aka an unspecified timeframe would match any and all legs, modulo the other matching predicates for that row.  Whether that ends up affecting the price will be a function of the other fare leg rule matching predicates, associated fare products, etc.

Does that make sense?

Reply:
I think so, but just to confirm: you're suggesting that we keep the initial design, correct? Which is:

field  | Description
--  | --
`from_timeframe_id`  | Defines a departure timeframe for the fare leg rule. If there are no matching `fare_leg_rules.from_timeframe_id` values to the timeframe_id being filtered, empty `fare_leg_rules.from_timeframe_id` will be matched by default. </br></br>**An empty entry in `fare_leg_rules.from_timeframe_id` corresponds to all timeframes defined in timeframes.timeframe_id excluding the ones listed under `fare_leg_rules.from_timeframe_id`**.
`to_timeframe_id` | Defines an arrival timeframe for the fare leg rule. If there are no matching `fare_leg_rules.to_timeframe_id` values to the `timeframe_id` being filtered, empty `fare_leg_rules.to_timeframe_id` will be matched by default. </br></br>**An empty entry in `fare_leg_rules.to_timeframe_id` corresponds to all timeframes defined in `timeframes.timeframe_id` excluding the ones listed under `fare_leg_rules.to_timeframe_id`**.

Reply:
And if this is the case, how to represent that the time that matters for the peak fare is only when you start the leg, not when you end the leg?

Reply:
Should we differentiate these two cases?
- case 1: both `from_timeframe_id` and `to_timeframe_id` are empty -> interpreted as "corresponds to all timeframes [...] excluding [...]".
- case 2: one of them is specified and not the other -> interpreted as "the start/end time of the leg does not affect the fare"

Reply:
I wasn't arguing for keeping the original design.  I was arguing for where we ended up (as of March 24).  Namely:

> An empty `start|end_timeframe_group_id` indicates that the time of the leg does not affect the fare.

As such, I'd argue that we can resolve this conversation.